src/com/etendoerp/dependencymanager/util/DependencyTreeBuilder.java

```diff
@@ -130,4 +135,88 @@ private static boolean isBundle(PackageDependency dependency) {
     return dependency.getArtifact().contains(".extensions");
   }
 
-}
+  /**
+   * Recursively finds all sub-dependencies of the given dependency.
+   * This method returns a list of all sub-dependencies, excluding external dependencies and those with the artifact "ETENDO_CORE".
+   * It also updates the parent map with parent-child relationships.
+   *
+   * @param dependency
+   *     The dependency for which to find sub-dependencies.
+   * @param parentMap
+   *     A map to be updated with parent-child relationships.
+   * @return A list of sub-dependencies of the given dependency.
+   */
+  public static List<PackageDependency> searchSubDependency(PackageDependency dependency,
+      Map<String, String> parentMap) {
+    if (dependency.isExternalDependency()) {
+      return new ArrayList<>();
+    }
+
+    List<PackageDependency> dependencies = dependency.getDependencyVersion().getETDEPPackageDependencyList();
+    if (dependencies.size() == 1 && StringUtils.equals(ETENDO_CORE, dependencies.get(0).getArtifact())) {
+      return new ArrayList<>();
+    }
+    removeDependecyCore(dependencies);
+    List<PackageDependency> allDependencies = new ArrayList<>(dependencies);
+
+    for (PackageDependency dep : dependencies) {
+      parentMap.put(dep.getId(), dependency.getId());
+
+      List<PackageDependency> subDependencies = searchSubDependency(dep, parentMap);
+      allDependencies.addAll(subDependencies);
+    }
+    return allDependencies;
```

Comment:
It could be that in this list, there are repeated items? because maybe the best thing would be to have a Set instead of a list to accumulate all the dependencies and before returning in the function, you convert it to a list...
Check if that makes sense, but it could be an improvement.

Reply:
You were right, I switched to a Set to avoid duplicates and then convert it to a List.